### PR TITLE
feat(curation): add Curation contract

### PR DESCRIPTION
**Pull request by robertu7**
**TODO**
* ~~make `Curation.sol` gasless compatiable;~~

### CODE REVIEW

File: src/Curation/Curation.sol [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    /// @inheritdoc ICuration
+    function curate(address creator_, string calldata uri_) public payable {
+        if (creator_ == address(0)) revert ZeroAddress();
+        if (msg.value <= 0) revert ZeroAmount();
+        if (bytes(uri_).length == 0) revert InvalidURI();
+        if (msg.sender == creator_) revert SelfCuration();
+
+        (bool success, ) = creator_.call{value: msg.value}("");
+        require(success);
```
Comment (gary02) [inline src/Curation/Curation.sol:41]: Why not  raise a concrete error?

**Comment by guoliu** [comment]
Also commented on notion: should we seperate `curate` and `curateFrom` API, to be consistent with ERC20?

**Comment by robertu7** [comment]
deployed to [0x94166663d4F8e985f164d335269CD64E4c8620cA](https://mumbai.polygonscan.com/address/0x94166663d4F8e985f164d335269CD64E4c8620cA) @gary02

**Comment by robertu7** [comment]
revised the `Curation` event and redeployed to [0xa219c6722008aa22828b31a13ab9ba93bb91222c](https://mumbai.polygonscan.com/address/0xa219c6722008aa22828b31a13ab9ba93bb91222c)

**Pull request merged**